Lib/fontTools/feaLib/parser.py

```diff
@@ -827,7 +827,13 @@ def parse_substitute_(self):
                 location)
 
         # GSUB lookup type 6: Chaining contextual substitution.
-        assert len(new) == 0, new
+        if len(new) != 0:
```

Comment:
this negative condition is a bit obscure. I had to look at the code above to understand what that means. Basically, if we haven't handled the sequence of glyphs after by/from yet by the time we reach the end of the parse_substitute_ method, it means there is something wrong with the sub rule, it's unknown or invalid. It doesn't necessarily means that one wanted to write a contextual substitution and forgot the apostrophe, but I can see how this may be frequent.
Maybe add a comment.

Reply:
thanks for the advice! I'll make the error message a bit less specific (remove the reference to contextual substitution).